soccer/Referee.cpp

```diff
@@ -110,103 +29,151 @@ static const bool CancelBallPlaceOnHalt = true;
 Referee::Referee(Context* const context)
     : stage_(NORMAL_FIRST_HALF_PRE),
       command_(HALT),
-      _running(false),
-      _context(context) {}
+      sent_time{},
+      received_time{},
+      stage_time_left{},
+      command_counter{},
+      yellow_info{},
+      blue_info{},
+      _kickDetectState{},
+      _readyBallPos{},
+      _kickTime{},
+      _mutex{},
+      _packets{},
+      _context(context),
+      prev_command_{},
+      prev_stage_{},
+      ballPlacementx{},
+      ballPlacementy{},
+      _network_thread(),
+      _asio_socket{_io_service},
+      _req_stop_prom{},
+      _stop_prom{},
+      _req_stop_fut{_req_stop_prom.get_future()},
+      _stop_fut{_stop_prom.get_future()} {}
 
 Referee::~Referee() { stop(); }
 
+void Referee::start() { _network_thread = std::thread{&Referee::run, this}; }
```

Comment:
I think you can honestly get rid of the network thread if you're on asio. Look at https://github.com/RoboJackets/robocup-software/blob/staging/soccer/radio/NetworkRadio.hpp for an example.